packages/api/cms-api/src/warnings/warning-event-subscriber.ts

```diff
@@ -53,15 +61,60 @@ export class WarningEventSubscriber implements EventSubscriber {
                 });
                 for (const node of flatBlocks.depthFirst()) {
                     const warnings = node.block.warnings();
-                    await this.warningService.updateWarningsForBlock(warnings, {
+
+                    const startDate = new Date();
+                    const sourceInfo = {
                         rootEntityName: entity.name,
                         rootColumnName: key,
                         targetId: args.entity.id,
                         rootPrimaryKey: args.meta.primaryKeys[0],
                         jsonPath: node.pathToString(),
+                    };
+                    await this.warningService.saveWarnings({
+                        warnings,
+                        type: "Block",
+                        sourceInfo,
                     });
+                    await this.warningService.deleteOutdatedWarnings({ date: startDate, type: "Block", sourceInfo });
+                }
+            }
+
+            const createWarnings = this.reflector.getAllAndOverride<CreateWarningsMeta>("createWarnings", [entity]);
+            if (createWarnings) {
+                const repository = this.entityManager.getRepository(entity);
+
+                const rows = await repository.find();
+
+                for (const row of rows) {
+                    let warnings: CreateWarningInput[] = [];
+                    if (this.isService(createWarnings)) {
+                        const service = this.moduleRef.get(createWarnings, { strict: false });
+
+                        if (service.createWarnings) {
+                            warnings = await service.createWarnings(row);
+                        }
```

Comment:
createWarnings is required
```suggestion
                        warnings = await service.createWarnings(row);
```

Reply:
changed in: 8c309cfd81935ae70337ed8bbe2d7d59d3e62ef9